writer.go

```diff
@@ -28,13 +28,16 @@ func NewWriter(w io.Writer) *Writer {
 }
 
 // Writer writes a single FEDWireMessage record to w
-func (w *Writer) Write(file *File) error {
+// options
+//  first bool : has variable length
+//  second bool : has not new line
+func (w *Writer) Write(file *File, options ...bool) error {
```

Comment:
This would be a good candidate for the functional Options pattern. Let's move these options up onto the `Writer` struct instead of accepting ordered boolean values here. Consider:
```go
type Writer struct {
	w              *bufio.Writer
	lineNum        int // current line being written
	variableLength bool
	noNewlines     bool
}

type OptionFunc func(*Writer)

func VariableLengthFields(variableLength bool) OptionFunc {
	return func(w *Writer) {
		w.variableLength = variableLength
	}
}

func NoNewlines(noNewlines bool) OptionFunc {
	return func(w *Writer) {
		w.noNewlines = noNewlines
	}
}

// NewWriter returns a new Writer that writes to w.
func NewWriter(w io.Writer, opts ...OptionFunc) *Writer {
	writer := &Writer{
		w: bufio.NewWriter(w),
	}

	for _, opt := range opts {
		opt(writer)
	}

	return writer
}

// Writer writes a single FEDWireMessage record to w
func (w *Writer) Write(file *File) error {
	if err := file.Validate(); err != nil {
		return err
	}
	w.lineNum = 0
	// Iterate over all records in the file
	if err := w.writeFEDWireMessage(file); err != nil {
		return err
	}
	w.lineNum++

	return w.w.Flush()
}
```

This way all methods on the `writer` have access to these flags without having to pass around arbitrary boolean parameters.